google-cloud-firestore/src/main/java/com/google/cloud/firestore/Firestore.java

```diff
@@ -56,14 +56,14 @@
   Iterable<CollectionReference> listCollections();
 
   /**
-   * Creates and returns a new @link{Query} that includes all documents in the database that are
-   * contained in a collection or subcollection with the given @code{collectionId}.
+   * Creates and returns a new {@link CollectionGroup} that includes all documents in the database
+   * that are contained in a collection or subcollection with the given @code{collectionId}.
    *
    * @param collectionId Identifies the collections to query over. Every collection or subcollection
    *     with this ID as the last segment of its path will be included. Cannot contain a slash.
    * @return The created Query.
    */
-  Query collectionGroup(@Nonnull String collectionId);
+  CollectionGroup collectionGroup(@Nonnull String collectionId);
```

Comment:
I was taking a look at the build failure for clirr and it correctly pointed out that this change is a binary breaking change. (It's source compatible, which would force people to re-compile their apps). I'm going to check with folks to see what our policy around this type of change is in a major version before adding the allow rule.

```
Exception in thread "main" java.lang.NoSuchMethodError: 'com.google.cloud.firestore.Query com.google.cloud.firestore.Firestore.collectionGroup(java.lang.String)'
        at io.github.benwhitehead.firestore.TestCollectionGroupSigChange.main(TestCollectionGroupSigChange.java:17)
```

Reply:
Do you have any updates on this? We have 11 days left to get this into our customers hand if we want to make the Aug 15 deadline :)

Reply:
Chatting about it this afternoon

Reply:
This is a breaking change, IMHO

Reply:
This is going out in a major release. I will add "!" to the PR title.

Reply:
I just pushed the updated clirr rules to cover this.